lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java

```diff
@@ -0,0 +1,179 @@
+            .buildAndStore(taskService);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_DeleteTaskCommentAndUpdateNumberOfComments_For_TaskCommentId() throws Exception {
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(0);
+    assertThat(taskService.getTask(task1.getId()).getNumberOfComments()).isZero();
+  }
+
+  @WithAccessId(user = "user-1-2", groups = "user-1-1")
+  @Test
+  void should_FailToDeleteTaskComment_When_UserHasNoAuthorization() throws Exception {
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(task1.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService, "user-1-1");
+    ThrowingCallable call = () -> taskService.deleteTaskComment(comment1.getId());
+    NotAuthorizedOnTaskCommentException e =
+        catchThrowableOfType(call, NotAuthorizedOnTaskCommentException.class);
+    assertThat(e.getTaskCommentId()).isEqualTo(comment1.getId());
+    assertThat(e.getCurrentUserId()).isEqualTo(taskanaEngine.getCurrentUserContext().getUserid());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+    assertThat(taskService.getTask(task1.getId()).getNumberOfComments()).isOne();
+
+    taskanaEngine.runAsAdmin(
+        () -> {
+          try {
+            taskService.deleteTaskComment(comment1.getId());
+          } catch (TaskCommentNotFoundException ex) {
+            throw new SystemException("Cannot find task comment with id " + ex.getTaskCommentId());
+          } catch (TaskNotFoundException ex) {
+            throw new SystemException("Cannot find task comment with id " + ex.getTaskId());
+          } catch (NotAuthorizedOnTaskCommentException ex) {
+            throw new SystemException(
+                "User "
+                    + ex.getCurrentUserId()
+                    + " not authorized on task comment "
+                    + ex.getTaskCommentId());
+          } catch (NotAuthorizedOnWorkbasketException ex) {
+            throw new SystemException(
+                "User "
+                    + ex.getCurrentUserId()
+                    + " not authorized on workbasket "
+                    + ex.getWorkbasketId());
+          }
+        });
+    List<TaskComment> taskCommentsAfterDeletionWithAdmin =
+        taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletionWithAdmin).hasSize(0);
```

Comment:
btw: what is the difference between isZero and hasSize(0)?

Reply:
isZero checks if something is 0, hasSize(0) checks if a list has the size 0. hasSize(0) and isEmpty() does the same thing. I will change it to isEmpty() because it is easier to read.